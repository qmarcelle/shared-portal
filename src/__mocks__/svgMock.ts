import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
