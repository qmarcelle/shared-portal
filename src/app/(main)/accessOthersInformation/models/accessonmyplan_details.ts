export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
