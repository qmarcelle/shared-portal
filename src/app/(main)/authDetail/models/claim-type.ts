export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
