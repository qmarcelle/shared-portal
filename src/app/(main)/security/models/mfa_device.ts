import { MfaDeviceType } from './mfa_device_type';

export interface MfaDevice {
  deviceType: MfaDeviceType;
  enabled: boolean;
  label: string;
  subLabel?: string;
  emailOrPhone: string | null;
  keyUri?: string;
  secret?: string;
}
