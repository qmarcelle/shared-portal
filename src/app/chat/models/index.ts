// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
