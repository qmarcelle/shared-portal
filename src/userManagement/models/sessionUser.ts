import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
