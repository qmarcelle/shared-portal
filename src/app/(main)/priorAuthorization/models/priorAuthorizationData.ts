import { ClaimDetails } from '../../../../models/claim_details';

export const priorAuthorizationData: ClaimDetails[] = [
  {
    id: 'Claim1',
    claimStatus: 'Pending',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Physical Therapy',
    memberName: 'Chris Hall',
    serviceDate: '04/06/23',
    totalBilled: '',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Anand Patel',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim2',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Magnetic Resonance Imaging (MRI)',
    memberName: 'Chris Hall',
    serviceDate: '01/23/23',
    totalBilled: '145.00',
    planPaid: '114.76',
    myShare: '30.24',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim3',
    claimStatus: 'Partial Approval',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Continuous Positive Airway Pressure Machiâ€¦',
    memberName: 'Chris Hall',
    serviceDate: '08/10/23',
    totalBilled: '403.98',
    planPaid: '338.37',
    myShare: '65.61',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim4',
    claimStatus: 'Denied',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Home Health Services',
    memberName: 'Chris Hall',
    serviceDate: '07/05/23',
    totalBilled: '403.98',
    planPaid: '0.00',
    myShare: '403.98',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Home Health Services',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim5',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Brittant Johnson',
    memberName: 'Chris Hall',
    serviceDate: '07/04/23',
    totalBilled: '183.00',
    planPaid: '150.85',
    myShare: '32.15',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
];
