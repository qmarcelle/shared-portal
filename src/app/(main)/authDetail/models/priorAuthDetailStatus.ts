import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
