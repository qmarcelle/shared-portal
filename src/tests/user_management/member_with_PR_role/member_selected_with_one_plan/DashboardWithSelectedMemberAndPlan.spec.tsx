import DashboardPage from '@/app/dashboard/page';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

// LoggedInUserInfo call
mockedAxios.get.mockResolvedValueOnce({
  data: {
    groupData: {
      groupID: '100000',
      groupCK: '21908',
      groupName: 'BlueCross BlueShield of Tennessee',
      parentGroupID: '100001',
      subGroupID: '0001',
      subGroupCK: 28951,
      subGroupName: 'BlueCross BlueShield of Tennessee',
      clientID: 'EI',
      policyType: 'INT',
      groupEIN: '620427913',
    },
    networkPrefix: 'QMI',
    subscriberID: '902218823',
    subscriberCK: '91722400',
    subscriberFirstName: 'CHRIS',
    subscriberLastName: 'HALL',
    coverageTypes: [
      {
        productType: 'M',
        coverageLevel: 'A',
        exchange: false,
        indvGroupInd: '',
        pedAdultInd: '',
      },
      {
        productType: 'D',
        coverageLevel: 'A',
        exchange: true,
        indvGroupInd: 'Group',
        pedAdultInd: 'Adult',
      },
      {
        productType: 'V',
        coverageLevel: '*',
        exchange: true,
        indvGroupInd: 'Group',
        pedAdultInd: 'Adult',
      },
    ],
  },
});
mockedAxios.get.mockResolvedValueOnce({
  data: {
    physicianId: '3118777',
    physicianName: 'Louthan, James D.',
    address1: '2033 Meadowview Ln Ste 200',
    address2: '',
    address3: '',
    city: 'Kingsport',
    state: 'TN',
    zip: '376607432',
    phone: '4238572260',
    ext: '',
    addressType: '1',
    taxId: '621388079',
  },
});
mockedAxios.get.mockResolvedValueOnce({
  data: {
    currentPolicies: [
      {
        memberCk: '502622001',
        subscriberName: 'REBEKAH WILSON',
        groupName: 'Radio Systems Corporation',
        memberId: '90447969100',
        planTypes: ['M', 'R', 'S'],
        amplifyMember: false,
      },
    ],
    pastPolicies: [],
  },
});
jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        id: 'testUser',
        currUsr: {
          umpi: '57c85test3ebd23c7db88287',
          role: UserRole.MEMBER,
          plan: {
            fhirId: '654543434',
            grgrCk: '7678765456',
            grpId: '65654323',
            memCk: '502622001',
            sbsbCk: '5654566',
            subId: '56543455',
          },
        },
        vRules: {},
      },
    }),
  ),
}));

describe('Dashboard when PR is selected', () => {
  let containerSnap: HTMLElement;
  beforeAll(async () => {
    const component = await DashboardPage();
    const { container } = render(component);
    containerSnap = container;
  });

  it('should render UI correctly with proper greeting and plan details', () => {
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/byMemberCk/502622001',
    );
    expect(screen.getByText('Welcome, Chris')).toBeVisible();
    expect(
      screen.getByText('Plan: BlueCross BlueShield of Tennessee'),
    ).toBeVisible();
    expect(screen.getByText('Policies: Medical, Dental, Vision')).toBeVisible();
    expect(screen.getByText('View Plan Details')).toBeVisible();

    expect(containerSnap).toMatchSnapshot();
  });
});
