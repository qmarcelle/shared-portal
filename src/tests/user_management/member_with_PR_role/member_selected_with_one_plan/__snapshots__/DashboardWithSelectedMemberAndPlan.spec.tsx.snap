// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard when PR is selected should render UI correctly with proper greeting and plan details 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center page"
  >
    <div
      class="flex flex-col h-175px w-full text-white justify-center items-center brand-gradient px-4 "
    >
      <div
        class="flex flex-col items-start w-full app-content px-4"
      >
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <h1
          class="title-1 font-thin"
        >
          Welcome, 
           
          Chris
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Plan: Chris B Hall Enterprises
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Subscriber ID: 902218823
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Group ID: 100000
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Policies: Medical, Wellness, Vision, Dental
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <a
          class="link-white-text"
          href="/member/myplan"
        >
          <p
            class="pb-2 pt-2"
          >
            View Plan Details
          </p>
        </a>
      </div>
    </div>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <div
      class="flex flex-col w-full justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color"
      >
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main large-section"
            >
              <div
                class="flex flex-col"
              >
                <h2
                  class="title-2"
                >
                  Recent Claims
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="recentClaimsException"
                >
                  <div
                    class="flex flex-row p-2"
                  >
                    <img
                      alt=""
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                    <p
                      class="body-1 inline my-auto ml-4 p-2"
                      style="display: block; max-width: 100%;"
                    >
                      No Claims Found.
                    </p>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <a
                  aria-label="View Claims"
                  href="/claims"
                  style="display: block;"
                >
                  <button
                    class="flex flex-row link-container undefined"
                    style="max-width: max-content; height: auto; display: block;"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="link undefined"
                    >
                      View Claims
                    </p>
                  </button>
                </a>
              </div>
            </div>
            <div
              class="card-main large-section"
            >
              <div
                class="flex flex-col"
              >
                <h2
                  class="title-2"
                >
                  Medical & Pharmacy Balance
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="card-main neutral container undefined"
                >
                  <div
                    class="flex flex-row flex flex-row align-top m-4"
                  >
                    <img
                      alt=""
                      class="icon"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                    <span
                      style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-row flex flex-col flex-grow"
                    >
                      <p
                        class="body-1"
                      >
                        There was a problem loading your information. Please try refreshing the page or returning to this page later.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col  flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main large-section"
            >
              <div>
                <h2
                  class="title-2"
                >
                  Prior Authorization
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="card-elevated cursor-pointer"
                  tabindex="0"
                >
                  <div
                    class=" flex-row align-top m-4"
                  >
                    <div
                      class="flex"
                    >
                      <img
                        alt=""
                        class="icon"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                      <span
                        style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                      />
                      <span
                        class="font-bold"
                      >
                        Magnetic Resonance Images(MRI)
                      </span>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-col items-start"
                    >
                      <div
                        class="success flex flex-row justify-center items-center px-2 py-1 rounded"
                      >
                        <div
                          class="circle rounded-full w-3 h-3 aspect-square"
                        />
                        <p
                          class="font-bold ml-2 body-2"
                        >
                          Approved
                        </p>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                      />
                      <span
                        class="body-1"
                      >
                        Visited on 
                        01/23/23
                        , For 
                        Chris Hall
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <a
                  aria-label="View Prior Authorizations"
                  style="display: block;"
                >
                  <button
                    class="flex flex-row link-container undefined"
                    style="max-width: max-content; height: auto; display: block;"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="link undefined"
                    >
                      View Prior Authorizations
                    </p>
                  </button>
                </a>
              </div>
            </div>
          </div>
        </section>
        <h2
          class="title-2 !font-light !text-[32px]/[40px]"
        >
          Find Care & Costs
        </h2>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main large-section"
            >
              <div>
                <h3
                  class="title-2"
                >
                  Benefits & Coverage
                </h3>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="flex"
                >
                  <label
                    class="body-1"
                  >
                    Browse your benefits by category:
                  </label>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                  />
                </div>
                <div>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                  <a
                    class="flex flex-row w-fit"
                    href="/member/myplan/benefits"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <button
                        class="body-bold"
                      >
                        Other Benefits
                      </button>
                      <span
                        style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                      />
                    </div>
                    <div
                      class="flex flex-col "
                    >
                      <img
                        alt=""
                        class="icon mt-2"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                  </a>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <a
                  aria-label="View Benefits & Coverage"
                  style="display: block;"
                >
                  <button
                    class="flex flex-row link-container undefined"
                    style="max-width: max-content; height: auto; display: block;"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="link undefined"
                    >
                      View Benefits & Coverage
                    </p>
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col page-section-36_67 items-stretch"
          >
            <div
              class="card-main large-section"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[50px] h-[50px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for Care? Find A:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                />
              </section>
            </div>
            <div
              class="card-main large-section"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[50px] h-[50px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? You can estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                />
              </section>
            </div>
          </div>
        </section>
        <section />
        <section />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
    </div>
  </div>
</div>
`;
